### ISSUE TITLE
Grid config to set ctrl+c keyboard shortcut to cell (vs. row) copy

### ISSUE THREAD

**Issue by amcclain** [xh/hoist-react#1451]
Should default to true when `showCellFocus` true.

**Comment by lbwexler** [comment]
Discussed with Peter.  One subtlety is that the context menu labelling should also change.

**Comment by pauldelano** [comment]
Currently I'm implementing cell-range selection for copy. I added these settings to the grid config. By default ^C copies the range instead of the rows. Also, the Copy and Copy with Headers does so for the range instead of the visible grid.

agOptions: {
    enableRangeSelection: true,
    processCellForClipboard: ({value}) => {
        if (isDate(value)) {
            return // proper date formatting
        }
        return value;
    }
}

Can we get range copying built in?

**Comment by peterjcoffman** [comment]
I think we should look at what Paul's change does and then decide what the proper behavior is. Range selection is basically the same as this change but substantially more powerful (and intuitive).

**Comment by lbwexler** [comment]
Thanks for the input paul -- seems like you are raising a couple of different issues with your code-snippet here:

1) The fact that you needed to override `processCellForClipboard` to get the date formatted properly.  Would be very nice if out of the box, the user got the right formatting here, or at least
if there were a cleaner way for an application to specify what was desired here.  This relates very 
closely to how we control rendering in export, and I imagine in most cases, users would expect to get the same result for all cells, whether they exported them in to a spreadsheet, or copied them in to a spreadsheet.

2) Enabling 'Range Selection' as an option.  We need to know how to do this in a way that is compatible with our selectionModel property on GridModel, and our keyboard handling extensions to GridModel.

In short, I think there are perhaps two different new tickets that we could create here.

**Comment by amcclain** [comment]
Just catching up with this ticket - I see, as always, that it's more complicated than I realized. Mainly in that our current "Copy Cell" context menu item doesn't actually help us much when it comes to the keyboard shortcut. The ctx. menu item is relying on the fact that it has a row/column context.  We also now see that the current Copy Cell option is less-than-ideal in that it naively assumes you can copy `record[field]` directly.

Re. `processCellForClipboard` note that we provide a default impl in Grid that seems like it could get smarter: https://github.com/xh/hoist-react/blob/8d465c13cc43ca2ee01f241beffa599fa06bc40f/cmp/grid/Grid.js#L664

I think that linking the output of this function to `column.exportFormat` makes sense - that's how you expect the value to "exit" the grid in general - but we have the complication that those formats are actually an Excel formatting API....

So I guess we should meet - if you have not already - and talk through some of these concerns (and range selection) when we can clear out an hour or so. Looking this over makes me realize we have a few important ways we need to improve our copying story beyond this particular ticket.

**Comment by amcclain** [comment]
^ remind me to never use that label 😜

**Comment by haynesjm42** [comment]
Will take another look at this as I think this will be important for a good user experience in #883
